### PR TITLE
fix `useConfig`/`useRouter` inside `head()`

### PR DESCRIPTION
**Pull request by B2o5T**
fixes https://github.com/shuding/nextra/issues/704  https://github.com/shuding/nextra/issues/679

### CODE REVIEW

**Comment by shuding** [comment]
The potential solution I was thinking for #679 is we don't render `<Head/>` but instead call it directly in the layout component:

```jsx
const head = Head()
return <NextHead>{head}</NextHead>
```

**Comment by B2o5T** [comment]
@shuding i guess it’s will still not works, I remember that I tested it

**Comment by B2o5T** [comment]
Yeah, do you mean inside `head`? If yes, this should works!

**Comment by B2o5T** [comment]
@shuding finally fixed! so now hooks can be used as well inside `head()` but return should be wrapped with `<NextHead />`
image
image

File: .changeset/sweet-worms-visit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'nextra-theme-docs': patch
+---
+
+fix `useConfig`/`useRouter` inside `head()`, should be wrapped with `NextHead />`
```
Comment (shuding) [inline .changeset/sweet-worms-visit.md:5]:
```suggestion
fix `useConfig`/`useRouter` inside `head()`, should be wrapped with `<NextHead/>`
```

File: packages/nextra-theme-docs/src/components/head.tsx [hunk 2]
```diff
@@ -12,38 +11,41 @@ export function Head(): ReactElement {
...
+            />
+          </>
+        )}
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1.0, viewport-fit=cover"
+        />
+      </NextHead>
+      {/* since in NextHead we can't use hooks */}
+      {renderComponent(config.head)}
```
Comment (shuding) [inline packages/nextra-theme-docs/src/components/head.tsx:48]: My original idea was to do this:

```suggestion
        {config.head()}
      </NextHead>
```

So it's required to be a functional component.

Comment (B2o5T) [reply packages/nextra-theme-docs/src/components/head.tsx]: nice, now it's works 💪. config and router no longer empty
image

File: .changeset/sweet-worms-visit.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'nextra-theme-docs': patch
+---
+
+fix `useConfig`/`useRouter` inside `head()`, should be wrapped with `<NextHead/>`
```
Comment (B2o5T) [inline .changeset/sweet-worms-visit.md:5]:
```suggestion
fix `useConfig`/`useRouter` inside `head()`
```

File: packages/nextra-theme-docs/src/components/head.tsx (hunk 2 above)
Comment (shuding) [reply packages/nextra-theme-docs/src/components/head.tsx]: Thanks!!

**Pull request merged**